samples/DispatcherSample.Web/StartupUsingLinkGenerator.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+
+namespace DispatcherSample.Web
+{
+    /// <summary>
+    /// This sets up a sample where a custom EndpointDataSource creates endpoints with MethodInfo based address
+    /// and links are generated based on the lookup of this MethodInfo.
+    /// Flow of a request is: Dispatcher selects an endpoint => Endpoint middleware is invoked => Selected endpoint
+    /// invokes a func which creates an invoker called IHubInvoker which uses the current endpoint's MethodInfo to
+    /// instantiate and execute the target method.
+    /// </summary>
+    public class StartupUsingLinkGenerator
```

Comment:
Discussed in person, and this was just done as the thought experiment, and not to be checked in.